src/parallel/communicator.C

```diff
@@ -170,6 +177,15 @@ void Communicator::assign(const communicator & comm)
 
       libmesh_assert_greater_equal (i, 0);
       _rank = cast_int<processor_id_type>(i);
+
+      // Get the maximum tag value
+      {
+        MPI_Aint * maxTag;
+        int flag;
+        libmesh_call_mpi(MPI_Comm_get_attr(MPI_COMM_WORLD, MPI_TAG_UB, &maxTag, &flag));
+
+        _max_tag = *maxTag;
```

Comment:
This appears to be incorrect.  On MPICH 3.2 I get -4026531841 this way, which doesn't fit in int and isn't even positive.  The MPI_Comm_get_attr man page seems to think we need to be passing in a ```void*``` and then casting the pointer to ```int*``` before dereferencing.

I'll do that; I've got a fork of this branch in which I'm trying out my "see if I can retain backwards tagvalue compatibility without breaking Derek's stuff" plan.